File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

#include <math.h>

bool is_armstrong_number(int candidate) {
    int tmp = candidate;
    int power = 0;
    while (tmp > 0) {
        tmp /= 10;
        power += 1;
    }
    if (power == 0) {
        power += 1;
    }
    int sum = 0;
    tmp = candidate;
    while (tmp > 0) {
        int d = tmp % 10;
        tmp /= 10;
        sum += pow(d, power);
    }
    return sum == candidate;
}

File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
import math

def is_armstrong_number(number) -> bool:
    if number == 0:
        return True

    digits_num = math.ceil(math.log(number, 10))
    sum = 0
    tmp = number
    for _ in range(0, digits_num):
        d = tmp % 10
        tmp //= 10
        sum += d ** digits_num
    return sum == number